Add tracing methods again

Uxio0 opened the pull request:
### What was wrong?

Closes #2830

### How was it fixed?

Add tracing methods again

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

### Cute animal picture

![](https://avatars.githubusercontent.com/u/6909403?v=4)

Uxio0 committed:
Add tracing methods again

- Add them on `web3.tracing`
- Closes #2830

fselmo reviewed (commented):
Thanks @Uxio0. I agree these are useful and it sits much nicer in a `tracing` module. `v6` is out now so this can probably start inching its way back in. I just made a quick pass and added some general cleanup notes / nits. Would be good to add some tests before we merge this in. @kclowes I can try to contribute here this week and see if we can pull some from the old `Parity` module code (though that might take a bit to set up proper tests).

Uxio0 committed:
Rename missing Parity tracing types

Uxio0 committed:
Remove uneeded docstring

fselmo committed:
Lint fixes and docs fix

fselmo committed:
Add formatters to ``tracing`` module; make default

- Add formatters for ``tracing`` module methods for human-readable values.
- Make the ``tracing`` module a default module on the ``Web3`` class instance.

fselmo committed:
Refactor + consolidate transaction request formatters

- Remove `maxFeePerGas` and `maxPriorityFeePerGas` request formatters from `method_formatters.py` and add them in `rpc_abi.py` along with the rest of the transaction request formatters for consistency. This separation just causes confusion.
- Use ``TRANSACTION_PARAMS_ABIS`` formatters for ``trace_call``.

fselmo committed:
Newsfragment for 2851

pacrob reviewed (approved):
lgtm!

fselmo committed:
Add trace_filter formatters
